mobile/lib/shared/ui/immich_image.dart

```diff
@@ -89,37 +91,36 @@ class ImmichImage extends StatelessWidget {
     }
     final String? token = Store.get(StoreKey.accessToken);
     final String thumbnailRequestUrl = getThumbnailUrl(asset, type: type);
-    return CachedNetworkImage(
-      imageUrl: thumbnailRequestUrl,
-      httpHeaders: {"Authorization": "Bearer $token"},
-      cacheKey: getThumbnailCacheKey(asset, type: type),
+    Widget placeholderWidget;
+    if (asset.thumbhash != null) {
+      placeholderWidget = FittedBox(
+        fit: BoxFit.fill,
+        child: Image(
+          image: ThumbHash.fromIntList(asset.thumbhash!).toImage(),
+        ),
+      );
+    } else if (useGrayBoxPlaceholder) {
+      placeholderWidget = const DecoratedBox(
+        decoration: BoxDecoration(color: Colors.grey),
+      );
+    } else {
+      placeholderWidget = Transform.scale(
+        scale: 0.2,
+        child: const CircularProgressIndicator(),
+      );
+    }
+
+    return ImageFade(
       width: width,
       height: height,
-      // keeping memCacheWidth, memCacheHeight, maxWidthDiskCache and
-      // maxHeightDiskCache = null allows to simply store the webp thumbnail
-      // from the server and use it for all rendered thumbnail sizes
+      image: CachedNetworkImageProvider(
+        thumbnailRequestUrl,
+        headers: {"Authorization": "Bearer $token"},
+        cacheKey: getThumbnailCacheKey(asset, type: type),
+      ),
       fit: fit,
-      fadeInDuration: const Duration(milliseconds: 250),
-      progressIndicatorBuilder: (context, url, downloadProgress) {
-        if (useGrayBoxPlaceholder) {
-          return const DecoratedBox(
-            decoration: BoxDecoration(color: Colors.grey),
-          );
-        }
-        return Transform.scale(
-          scale: 0.2,
-          child: CircularProgressIndicator.adaptive(
-            value: downloadProgress.progress,
-          ),
-        );
-      },
-      errorWidget: (context, url, error) {
-        if (error is HttpExceptionWithStatus &&
-            error.statusCode >= 400 &&
-            error.statusCode < 500) {
-          debugPrint("Evicting thumbnail '$url' from cache: $error");
-          CachedNetworkImage.evictFromCache(url);
-        }
+      placeholder: placeholderWidget,
+      errorBuilder: (context, error) {
```

Comment:
This now works fine with no flicker. The main tradeoff I made here is that I had to use `CachedNetworkImageProvider` instead of `CachedNetworkImage` so error handling had to be moved up the chain (and worsened to some degree). Input on how to make this better would be appreciated. 

Besides that I think all is now well.

Reply:
I tried this and it looks good.

Can you put
```dart
if (error is HttpExceptionWithStatus &&
    error.statusCode >= 400 &&
    error.statusCode < 500) {
  debugPrint("Evicting thumbnail '$thumbnailRequestUrl' from cache: $error");
  CachedNetworkImage.evictFromCache(thumbnailRequestUrl);
}
```
into the `errorBuilder`?

Reply:
Ah okay didn't know if it worked like that thanks.